src/models/pocketSave.ts

```diff
@@ -80,22 +80,72 @@ export class PocketSaveModel {
       uniqueIds,
       timestamp
     );
-    const errors =
-      missing.length > 0
-        ? missing.map((missingId) => this.notFoundPayload('id', missingId))
-        : [];
-    const save = updated.map((row) => PocketSaveModel.transformListRow(row));
+    const { save, resolvedErrors } = this.fetchSavesAndError(
+      missing,
+      updated,
+      path
+    );
     // TODO: Emit events
     // save.forEach((saveItem) => {
     //   this.context.emitItemEvent(EventType.ARCHIVE_ITEM, saveItem);
     // });
     // Hydrate errors path with current location
     // Resolved on saveArchive because it needs to be aware
     // of this path, not the path of the error resolver type
+    return { save, errors: resolvedErrors };
+  }
+
+  /**
+   * bulk favorite the saves
+   * no action performed if saves are already favorited
+   * any error in the batch will rollback and fail the entire batch
+   * @param ids itemIds associated to the saves to archive; must all be
+   * valid or the operation will fail.
+   * @param timestamp timestamp for when the bulk operation occurred
+   * @returns an array of updated saves and the missing ids (if any
+   * encountered; these arrays are mutually exclusive, so if one is
+   * populated, the other is empty)
+   */
+  public async saveFavorite(
+    ids: string[],
+    timestamp: Date,
+    path: GraphQLResolveInfo['path']
+  ): Promise<SaveWriteMutationPayload> {
+    const uniqueIds = uniqueArray(ids.map((id) => parseInt(id)));
+    const { updated, missing } = await this.saveService.favoriteListRow(
+      uniqueIds,
+      timestamp
+    );
+    const { save, resolvedErrors } = this.fetchSavesAndError(
+      missing,
+      updated,
+      path
+    );
+    //todo: emit event
+    return { save, errors: resolvedErrors };
+  }
+
+  /**
+   * construct saves and errors types from the data services
+   * @param missing missing savesId
+   * @param updated updated saves
+   * @param path mutation path
+   * @private
+   */
+  private fetchSavesAndError(
+    missing: string[],
+    updated: ListResult[],
+    path: GraphQLResolveInfo['path']
+  ) {
+    const errors =
+      missing.length > 0
+        ? missing.map((missingId) => this.notFoundPayload('id', missingId))
+        : [];
+    const save = updated.map((row) => PocketSaveModel.transformListRow(row));
     const resolvedErrors = errors.map((error) => ({
       ...error,
       path: this.context.models.baseError.path(path),
     }));
-    return { save, errors: resolvedErrors };
+    return { save, resolvedErrors };
```

Comment:
This is for the mutation payload response type

```suggestion
    return { save, errors: resolvedErrors };
```